src/main/java/se/liu/rtslab/energybox/MainFormController.java

```diff
@@ -1,559 +1,566 @@
+    // -> "DNS" - a phone is usually not running a DNS server so any requests are from the the recording device
+    String sourceIP = "";
+    HashMap<String, Integer> addressOccurrence = new HashMap();
+    boolean error = false;
+    String os;
+    
+    final ObservableList<Packet> packetList = FXCollections.observableList(new ArrayList());
+    @FXML
+    private Button modelButton;
+    @FXML
+    private Button deviceButton;
+    @FXML
+    private Button networkButton;
+    @FXML
+    private Button traceButton;
+    @FXML
+    private ImageView image;
+    
+    @Override
+    public void initialize(URL url, ResourceBundle rb)
+    {
+        // Load the icon for the Model button
+        try {
+            image.setImage(new Image("img/gears.png", true));
+        } catch (Exception e) {
+            System.err.println("Gears image not found");
+        }
+        
+        // Checks between the two supported operating systems and tries to add
+        // the directory where the JAR file is locted to the PATH or CLASSPATH
+        // variables in the JVM.
+        // OS X uses tshark instead, so the process is different.
+        os = OSTools.getOS();
+        switch(os)
+        {
+            case "Windows":
+            {
+                String location = MainFormController.class.getProtectionDomain().getCodeSource().getLocation().getPath();
+                String decodedLocation = "";
+                try { decodedLocation = URLDecoder.decode(location, "UTF-8"); }
+                catch (UnsupportedEncodingException e){ e.printStackTrace(); }
+                StringBuilder relativePath = new StringBuilder();
+                relativePath.append(new File(decodedLocation).getParent());
+                relativePath.append(File.separator);
+                try { OSTools.addDirectory(relativePath.toString()); }
+                catch (IOException e)
+                {
+                    OSTools.showErrorDialog("JVM path error!", e.getMessage());
+                }
+            }
+            break;
+                
+            case "Linux":
+            {
+                String location = MainFormController.class.getProtectionDomain().getCodeSource().getLocation().getPath();
+                String decodedLocation = "";
+                try { decodedLocation = URLDecoder.decode(location, "UTF-8"); }
+                catch (UnsupportedEncodingException e){ e.printStackTrace(); }
+                StringBuilder relativePath = new StringBuilder();
+                relativePath.append(new File(decodedLocation).getParent());
+                relativePath.append(File.separator);
+                try { OSTools.addDirectory(relativePath.toString()); }
+                catch (IOException e)
+                {
+                    OSTools.showErrorDialog("JVM Path Error", e.getMessage());
+                }
+                try { WinPcap.isSupported(); }
+                catch (UnsatisfiedLinkError e)
```

Comment:
If the idea is to switch Linux to tshark by default, then this should probably be `catch (UnsatisfiedLinkError|NoClassDefFoundError e)` since the jnetpcap.jar file wouldn't be bundled.